src/test/java/com/github/cameltooling/lsp/internal/telemetry/TelemetryTest.java

```diff
@@ -36,4 +42,26 @@ void testInitializationWithJvmInfoMetric() throws Exception {
 		assertThat((String)properties.get("jvm.version")).isNotBlank();
 	}
 	
+	@ParameterizedTest(name= "Telemetry sent for language {1}")
+	@MethodSource
+	void testOpenedDocumentMetric(String content, String extension) throws Exception {
+		initializeLanguageServer(content, "."+extension);
+		await("Await that telemetry event is sent as it is done async to not slow down the server")
+			.untilAsserted(() -> assertThat(telemetryEvents).hasSize(2));
+		TelemetryEvent telemetryEvent = telemetryEvents.get(1);
+		assertThat(telemetryEvent.name).isEqualTo(TelemetryManager.OPENED_DOCUMENT);
+		Map<String, Object> properties = telemetryEvent.properties;
+		assertThat((String)properties.get("language")).isEqualTo(extension);
+	}
+	
+	private static Stream<Arguments> testOpenedDocumentMetric() {
+		return Stream.of(
+				Arguments.of(RouteTextBuilder.createXMLSpringRoute(""), "xml"),
+				Arguments.of("camel", "java"),
```

Comment:
I thought again about it. This heuristic migth be not enough restrictive for Java file. it will send an event for all Java files that have camel in one of import. To be comparable with other files, it should be more restrictive as others are more precise on (almost) only Route files. Maybe to check that it also contains a `from`.

Reply:
might be better